test/exercise/arrays/solution.rb

```diff
@@ -1,12 +1,38 @@
 module Exercise
   module Arrays
     class << self
+      def max(array)
+        array.reduce { |acc, current | acc > current ? acc : current }
+      end
+
       def replace(array)
-        array
+        maximum = max(array)
+        array.map { |number| number > 0 ? maximum : number }
       end
 
-      def search(_array, _query)
-        0
+      def search(_array, _query, index = 0)
```

Comment:
Please check this file with rubocop. Leading underscore is usually used for unused variables.

Reply:
removed all leading underscores